modelon/impact/client/experiment_definition.py

```diff
@@ -226,14 +317,174 @@ def to_dict(self):
         """
         return {
             "experiment": {
-                "analysis": {
-                    "analysis_function": self.custom_function.name,
-                    "parameters": self.custom_function.parameter_values,
-                    "simulation_options": self._simulation_options,
-                    "solver_options": self._solver_options,
-                    "simulation_log_level": self.simulation_log_level,
+                "version": 2,
+                "base": {
+                    "model": {"fmu": {"id": self.fmu.id}},
+                    "modifiers": {"variables": self.variable_modifiers},
+                    "analysis": {
+                        "type": self.custom_function.name,
+                        "parameters": self.custom_function.parameter_values,
+                        "simulationOptions": self._simulation_options,
+                        "solverOptions": self._solver_options,
+                        "simulationLogLevel": self.simulation_log_level,
+                    },
                 },
-                "fmu_id": self.fmu.id,
-                "modifiers": {"variables": self.variable_modifiers},
+                "extensions": self.extensions,
             }
         }
+
+
+class BaseExperimentExtension(ABC):
+    """
+    Base class for an experiment extension class.
+    """
+
+
+class SimpleExperimentExtension(BaseExperimentExtension):
```

Comment:
I think better handling of default values is needed. If I do `SimpleExperimentExtension().with_modifiers({'a': 5}).to_dict()` I don't want anything related to `analysis`. Because if you do a lot of cases/extensions you want to send as little data as possible.